history/tools.py

```diff
@@ -83,39 +89,72 @@ def get_exchange_rate_btc_to_usd(d=None):
         d = datetime.datetime.now()
 
     exchange_pair = 'USDT_BTC'
-    latest_price = Price.objects.filter(symbol=exchange_pair,created_on__lt=d).order_by('-created_on').first()
+    latest_price = Price.objects.filter(symbol=exchange_pair, created_on__lt=d).order_by('-created_on').first()
     return latest_price.price
 
-def get_cost_basis(total_amount,symbol):
+
+def get_cost_basis(total_amount, symbol):
     from history.models import Trade
     running_amount = total_amount
     cost_basis_array = []
-    ts = Trade.objects.filter(symbol=symbol).order_by('-created_on').values_list('amount','price')
+    ts = Trade.objects.filter(symbol=symbol).order_by('-created_on').values_list('amount', 'price')
     for t in ts:
         if running_amount < 0:
             continue
         trade_amount = t[0]
         price = t[1]
         if running_amount < trade_amount:
-            cost_basis_array.append([running_amount,price])
+            cost_basis_array.append([running_amount, price])
         else:
-            cost_basis_array.append([trade_amount,price])
+            cost_basis_array.append([trade_amount, price])
         running_amount = running_amount - trade_amount
 
-    total_amount = sum([ t[0] for t in cost_basis_array])
-    total_paid = sum([ t[0] * t[1] for t in cost_basis_array])
+    total_amount = sum([t[0] for t in cost_basis_array])
+    total_paid = sum([t[0] * t[1] for t in cost_basis_array])
+
+    return total_paid / total_amount
 
-    return  total_paid / total_amount
 
 def get_deposit_balance():
     from history.models import Deposit
     btc_deposit_amount = 0
     usd_deposit_amount = 0
     for d in Deposit.objects.all():
-        exchange_rate = get_exchange_rate_to_btc(d.symbol,d.created_on)
+        exchange_rate = get_exchange_rate_to_btc(d.symbol, d.created_on)
         btc_amount = d.amount * exchange_rate
         usd_amount = get_exchange_rate_btc_to_usd(d.created_on) * btc_amount
         btc_deposit_amount = btc_deposit_amount + btc_amount
         usd_deposit_amount = usd_deposit_amount + usd_amount
     return btc_deposit_amount, usd_deposit_amount
 
+
+def npc(args):
```

Comment:
Can't you just do in models.py 342:
 ```python 
self._input = np.column_stack(self.ravel_args)
```

instead of:
```python
self._input = npc(self.ravel_args)
```
and remove the `npc `function altogether